nixos/modules/services/web-apps/part-db.nix

```diff
@@ -0,0 +1,247 @@
+
+    enableNginx = mkOption {
+      type = types.bool;
+      default = true;
+      description = ''
+        Whether to enable nginx or not. If enabled, an nginx virtual host will
+        be created for access to part-db. If not enabled, then you may use
+        `''${config.services.part-db.package}/public` as your document root in
+        whichever webserver you wish to setup.
+      '';
+    };
+
+    enablePostgresql = mkOption {
+      type = types.bool;
+      default = true;
+      description = ''
+        Whether to configure the postgresql database for part-db. If enabled,
+        a database and user will be created for part-db.
+      '';
+    };
+
+    virtualHost = mkOption {
+      type = types.str;
+      default = "localhost";
+      description = ''
+        The virtualHost at which you wish part-db to be served.
+      '';
+    };
+
+    poolConfig = lib.mkOption {
+      type = lib.types.attrsOf (
+        lib.types.oneOf [
+          lib.types.str
+          lib.types.int
+          lib.types.bool
+        ]
+      );
+      default = { };
+      defaultText = ''
+        {
+          "pm" = "dynamic";
+          "pm.max_children" = 32;
+          "pm.start_servers" = 2;
+          "pm.min_spare_servers" = 2;
+          "pm.max_spare_servers" = 4;
+          "pm.max_requests" = 500;
+        }
+      '';
+      description = ''
+        Options for the PartDB PHP pool. See the documentation on <literal>php-fpm.conf</literal>
+        for details on configuration directives.
+      '';
+    };
+
+    settings = lib.mkOption {
+      default = { };
+      description = ''
+        Options for part-db configuration. Refer to
+        <https://github.com/Part-DB/Part-DB-server/blob/master/.env> for
+        details on supported values. All <option>_FILE values supported by
+        upstream are supported here.
+      '';
+      example = lib.literalExpression ''
+        {
+          DATABASE_URL = "postgresql://db_user@localhost/db_name?serverVersion=16.6&charset=utf8&host=/var/run/postgresql";
+        }
+      '';
+      type = lib.types.submodule {
+        freeformType = lib.types.attrsOf (
+          with lib.types;
+          oneOf [
+            str
+            int
+            bool
+          ]
+        );
+        options = {
+          DATABASE_URL = lib.mkOption {
+            type = lib.types.str;
+            default = "postgresql://part-db@localhost/part-db?serverVersion=${config.services.postgresql.package.version}&host=/run/postgresql";
+            defaultText = "postgresql://part-db@localhost/part-db?serverVersion=\${config.services.postgresql.package.version}&host=/run/postgresql";
+            description = ''
+              The postgresql database server to connect to.
+              Defauls to local postgresql unix socket
+            '';
+          };
+        };
+      };
+    };
+  };
+
+  config =
+    let
+      phpEnv = cfg.phpPackage.buildEnv {
```

Comment:
And should we move that into the php package option?

Reply:
how? I tried to manipulate the default with buildEnv, but this didn't work

Reply:
We could use apply like https://github.com/NixOS/nixpkgs/blob/c3bb7450f2be4c01acc9d1b3e2ff45b7a21cd584/nixos/modules/services/misc/paperless.nix#L209